OWML.ModHelper.Menus/ModMessagePopup.cs

```diff
@@ -31,24 +30,32 @@ public void Initialize(PopupMenu popup)
             Menu = _twoButtonPopup;
         }
 
-        public IModMessagePopup Copy()
+        internal override void DestroySelf()
+        {
+            Object.Destroy(_twoButtonPopup);
+            OnConfirm = null;
+            OnCancel = null;
+            _twoButtonPopup = null;
+        }
+
+        internal ModMessagePopup Copy()
         {
             var newPopupObject = Object.Instantiate(_twoButtonPopup.gameObject);
             newPopupObject.transform.SetParent(_twoButtonPopup.transform.parent);
             newPopupObject.transform.localScale = _twoButtonPopup.transform.localScale;
+            newPopupObject.transform.localPosition = _twoButtonPopup.transform.localPosition;
             var newPopup = new ModMessagePopup(OwmlConsole);
             newPopup.Initialize(newPopupObject.GetComponent<PopupMenu>());
             return newPopup;
         }
```

Comment:
looks to me like DestroySelf and Copy can be moved into the base class.
See ModButton.Copy for an example of a base class instantiating a child of the child's type.
If you need to do some child-specific stuff, override Copy from base class, call base.Copy and do more stuff.

Reply:
actually, yeah, part of it can be moved. like add `protected GameObject Copy` in base, and move first halves of both methods to base